src/transaction_builder.cpp

```diff
@@ -141,6 +143,11 @@ void TransactionBuilder::AddSaplingSpend(
         throw std::runtime_error("TransactionBuilder cannot add Sapling spend to pre-Sapling transaction");
     }
 
+    // ZIP212: check that note plaintext lead byte is valid at height
+    if (!libzcash::plaintext_version_is_valid(consensusParams, nHeight, note.get_lead_byte())) {
```

Comment:
This shouldn't be checking using nHeight but rather nHeight + 1, because the transaction will appear in the next block.